### PR TITLE
feat(driver): option to force use of build for tests

### PR DESCRIPTION
**Pull request by unkrishnateja**

### CODE REVIEW

File: driver/driver.go [hunk 1]
```diff
@@ -146,7 +151,7 @@ func parseFlags() (*driverFlags, error) {
 	flags.commonFlags = cli.RegisterCommonFlags()
 	flag.StringVar(&flags.manualTestMode, "manual-test-mode", "skip", "How to handle affected tests tagged manual. Possible values: run|skip")
 	flag.StringVar(&flags.targetPatternFile, "target-pattern-file", "", "If defined, stores the list of affected targets in the given file.")
-
+	flag.StringVar(&flags.bazelCommandOverride, "bazel-command-override", "auto", "Overrides the bazel command which runs build/test based on target type to run only build. Options: build, default behaviour will selected build/test based on target type")
```
Comment (seh) [inline driver/driver.go:154]:
```suggestion
	flag.StringVar(&flags.bazelCommandOverride, "bazel-command-override", "auto", "Bazel subcommand used to build or test targets. Possible values: auto|build; by default, \"auto\" selects \"build\" or \"test\" based on the target's rule")
```

Comment (seh) [inline driver/driver.go:154]: Isn't this really overriding only the Bazel subcommand used for running tests? The flag name suggests that it's more general than that.

File: driver/driver.go [hunk 2]
```diff
@@ -28,6 +28,8 @@ type driverFlags struct {
 	targetPatternFile string
 	revisionBefore    string
 	manualTestMode    string
+	// Expected option: "build"
```
[the remark this answers is not in the document]
Comment (unkrishnateja) [reply driver/driver.go]: Yes, that is a more accurate term here, just updated, thank you.

File: driver/driver.go [hunk 3]
```diff
@@ -37,6 +39,8 @@ type config struct {
 	// One of "run" or "skip".
 	ManualTestMode    string
 	TargetPatternFile string
+	// Expected option: "build"
```
[the remark this answers is not in the document]
Comment (unkrishnateja) [reply driver/driver.go]: Yes, that is a more accurate term here, just updated, thank you.

File: driver/driver.go (hunk 1 above)
Comment (unkrishnateja) [reply driver/driver.go]: Thank you for the recommendation, updated.

Comment (unkrishnateja) [reply driver/driver.go]: I've changed this to bazelSubCommandOverride. Does this sound more accurate to you?

**Comment by unkrishnateja** [comment]
Thank you for the review @seh, please take a look at the changes.

File: driver/driver.go [hunk 4]
```diff
@@ -39,8 +39,8 @@ type config struct {
 	// One of "run" or "skip".
 	ManualTestMode    string
 	TargetPatternFile string
-	// Expected option: "build"
-	bazelCommandOverride string
+	// Expected value: "build"
```
Comment (seh) [inline driver/driver.go:42]: Should this be the following?
```suggestion
	// One of "auto" or "build".
```
The proposed implementation says that if the value is _not_ "build", we treat it as if it were "auto".

File: driver/driver.go [hunk 5]
```diff
@@ -28,8 +28,8 @@ type driverFlags struct {
 	targetPatternFile string
 	revisionBefore    string
 	manualTestMode    string
-	// Expected option: "build"
-	bazelCommandOverride string
+	// Expected value: "build"
```
Comment (seh) [inline driver/driver.go:31]: Likewise, should this refer to two possible values?

File: driver/driver.go [hunk 6]
```diff
@@ -28,8 +28,8 @@ type driverFlags struct {
...
+	bazelSubCommandOverride string
```
Comment (seh) [inline driver/driver.go:32]: Do you have ambition have this option do more than you're doing with it here? If not, you could name it something like "forceUseOfBuildForTests" and treat it as a boolean option, defaulting to false.

As named, it sounds like you're encouraging callers to supply a command name, but then you don't wind up actually using that command name unless it's "build".

Comment (unkrishnateja) [reply driver/driver.go]: Updated, thank you. I did start with the idea to allow the user to set either build/test, but chose to just force a build (I couldn't think of examples where one might want to force a test).

**Review by illicitonion — APPROVED** [review]
Works for me! Thanks!

**Comment by unkrishnateja** [comment]
@illicitonion Thanks! Is there someplace where I can see when the next release might be created?

**Comment by illicitonion** [comment]
I just released [v0.27.0](https://github.com/bazel-contrib/target-determinator/releases/tag/v0.27.0) 🎉  Thanks again!

**Comment by unkrishnateja** [comment]
Awesome, thank you!

**Pull request merged**